compiler/codegen/library.cpp

```diff
@@ -289,6 +291,42 @@ static void setupPythonTypeMap() {
 
 }
 
+static void setupFortranTypeMap() {
+  fortranKindNames[dtInt[INT_SIZE_8]->symbol] = "c_int8_t";
+  fortranKindNames[dtInt[INT_SIZE_16]->symbol] = "c_int16_t";
+  fortranKindNames[dtInt[INT_SIZE_32]->symbol] = "c_int32_t";
+  fortranKindNames[dtInt[INT_SIZE_64]->symbol] = "c_int64_t";
+
+  // No unsigned integers in Fortran.  Fake it with signed integers.
+  fortranKindNames[dtUInt[INT_SIZE_8]->symbol] = "c_int8_t";
+  fortranKindNames[dtUInt[INT_SIZE_16]->symbol] = "c_int16_t";
+  fortranKindNames[dtUInt[INT_SIZE_32]->symbol] = "c_int32_t";
+  fortranKindNames[dtUInt[INT_SIZE_64]->symbol] = "c_int64_t";
+
+  fortranKindNames[dtReal[FLOAT_SIZE_32]->symbol] = "c_float";
+  fortranKindNames[dtReal[FLOAT_SIZE_64]->symbol] = "c_double";
+  fortranKindNames[dtBool->symbol] = "c_bool";
+  //fortranKindNames[dtStringC->symbol] = "c_char";
+  fortranKindNames[dtComplex[COMPLEX_SIZE_64]->symbol] = "c_float_complex";
+  fortranKindNames[dtComplex[COMPLEX_SIZE_128]->symbol] = "c_double_complex";
+
+  fortranTypeNames[dtInt[INT_SIZE_8]->symbol] = "integer";
+  fortranTypeNames[dtInt[INT_SIZE_16]->symbol] = "integer";
+  fortranTypeNames[dtInt[INT_SIZE_32]->symbol] = "integer";
+  fortranTypeNames[dtInt[INT_SIZE_64]->symbol] = "integer";
+  fortranTypeNames[dtUInt[INT_SIZE_8]->symbol] = "integer";
+  fortranTypeNames[dtUInt[INT_SIZE_16]->symbol] = "integer";
+  fortranTypeNames[dtUInt[INT_SIZE_32]->symbol] = "integer";
+  fortranTypeNames[dtUInt[INT_SIZE_64]->symbol] = "integer";
```

Comment:
Does this mean that Fortran doesn't have different integer widths?

Reply:
The `fortranKindNames` map defined just above this is what specifies the width.  The generated interface for an argument `x: int(32)` would declare it as `integer(kind=c_int32_t), value :: x`.